File: 6.02/서지훈__프로그래머스_기지국_설치.java
class Solution {
    public int solution(int n, int[] stations, int w) {
        int answer = 0;

        double radio = 2 * w + 1;
        answer += (int)(Math.ceil((stations[0] - w - 1) / radio));

        for(int i = 0; i < stations.length-1; ++i){
            answer += (int)(Math.ceil((stations[i+1] - stations[i] - 2 * w - 1) / radio));
        }

        answer += (int)(Math.ceil((n - stations[stations.length-1] - w) / radio));

        return answer;
    }
}

File: 6.02/서지훈__프로그래머스_기지국_설치.py
import math

def solution(n, stations, w):
    answer = 0

    radio = 2 * w + 1
    answer += math.ceil((stations[0] - w - 1) / radio)

    length = len(stations)
    for i in range(length-1):
        answer += math.ceil((stations[i+1] - stations[i] - 2 * w - 1) / radio)
    answer += math.ceil((n - stations[length-1] - w) / radio)

    return answer